src/RA_Defs.h

```diff
@@ -12,28 +14,63 @@
 #include <deque>
 #include <map>
 
+
 #ifndef RA_EXPORTS
 
 //	Version Information is integrated into tags
 
 #else
 
+
+// Maybe an extra check just in-case
+
+#if _HAS_CXX17
+#define _DEPRECATED          [[deprecated]]
+#define _DEPRECATEDR(reason) [[deprecated(reason)]]
+#define _FALLTHROUGH         [[fallthrough]]//; you need ';' at the end
+#define _NODISCARD           [[nodiscard]]
+#define _NORETURN            [[noreturn]]
+#define _UNUSED              [[maybe_unused]]
+
+// Disables the use of const_casts, if you get an error, it's not a literal
+// type. You could use it on functions but they will need a deduction guide
+// That would probably be better with a forwarding function
```

Comment:
comment doesn't seem to apply to anything

Reply:
<code>inline constexpr auto</code> can sometimes be tedious to write out each time but if it's that important they can go away.